### PR TITLE
adding rehydration corral, which replaces a node in a given rke2 cluster

### PR DESCRIPTION
**Pull request by slickwarren**
This is a supporting PR for rancher automation: https://github.com/rancher/rancher/pull/39752

### CODE REVIEW

File: templates/aws-replace-node/terraform/pools/main.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  public_key = var.corral_public_key
+}
+
+resource "aws_instance" "replace" {
+  count = var.replace_count
+  ami = var.aws_ami
+  instance_type     = var.instance_type
+  key_name = aws_key_pair.corral_key.key_name
+  vpc_security_group_ids = [var.aws_security_group]
+  subnet_id = var.aws_subnet
```
Comment (jakefhyde) [inline templates/aws-replace-node/terraform/pools/main.tf:33]:
```suggestion
  count                  = var.replace_count
  ami                    = var.aws_ami
  instance_type          = var.instance_type
  key_name               = aws_key_pair.corral_key.key_name
  vpc_security_group_ids = [var.aws_security_group]
  subnet_id              = var.aws_subnet
```
Formatting, I generally use a terraform plugin for whatever IDE I'm using and it should format it correctly.

File: templates/rke2-replace-node/overlay/opt/corral/rke2/delete_nodes.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#1/bin/bash
```
Comment (jakefhyde) [inline templates/rke2-replace-node/overlay/opt/corral/rke2/delete_nodes.sh:1]:
```suggestion
#!/bin/bash
```
Also should be `delete-nodes.sh`

**Review by jakefhyde — CHANGES_REQUESTED** [review]
`.DS_STORE` files need to be removed, also some minor formatting issues. Also, the corral_node_pools should probably be updated once the node has been removed. This looks like it doesn't remove a node so much as delete it and add the same node back, is that intended? I imagine for rehydration we want to test with new nodes being added to the cluster.

**Comment by slickwarren** [comment]
spoke offline, the rehydration is resolved in the kubeconfig file as part of this package.

**Review by jakefhyde — APPROVED** [review]

**Review by sbstp — APPROVED** [review]

**Pull request merged**